File: src/test/java/com/owncloud/android/datamodel/OCFileTest.java
/*
 *
 * Nextcloud Android client application
 *
 * @author Tobias Kaminsky
 * Copyright (C) 2021 Tobias Kaminsky
 * Copyright (C) 2021 Nextcloud GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package com.owncloud.android.datamodel;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class OCFileTest {

    @Test
    public void testLongIds() {
        OCFile sut = new OCFile("/", "12345678ocjycgrudn78");
        assertEquals("12345678", sut.getLocalId());

        sut.setRemoteId("00000008ocjycgrudn78");
        assertEquals("8", sut.getLocalId());

        sut.setRemoteId("1234567891011ocjycgrudn78");
        assertEquals("1234567891011", sut.getLocalId());
    }

}

File: app/src/test/java/com/owncloud/android/datamodel/OCFileTest.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2022 Tobias Kaminsky <tobias@kaminsky.me>
 * SPDX-FileCopyrightText: 2022 Nextcloud GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.datamodel

import com.owncloud.android.lib.resources.shares.ShareType
import com.owncloud.android.lib.resources.shares.ShareeUser
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class OCFileTest {
    @Test
    fun testShareesDefaultsToEmptyMutableList() {
        val sut = OCFile("/")

        assertTrue(sut.sharees.isEmpty())

        sut.sharees.add(ShareeUser("alice", "Alice", ShareType.USER))
        assertEquals(1, sut.sharees.size)
    }

    @Test
    fun testShareesStayMutableWhenSetFromImmutableList() {
        val sut = OCFile("/")

        sut.sharees = listOf(ShareeUser("alice", "Alice", ShareType.USER))

        sut.sharees.add(ShareeUser("bob", "Bob", ShareType.USER))
        assertEquals(2, sut.sharees.size)
    }

    @Test
    fun testLongIds() {
        val sut = OCFile("/")

        sut.remoteId = "12345678ocjycgrudn78"
        assertEquals(12345678, sut.localId)

        sut.remoteId = "00000008ocjycgrudn78"
        assertEquals(8, sut.localId)

        // this will fail as fileId is too large
        sut.remoteId = "1234567891011ocjycgrudn78"
        assertNotEquals(1234567891011L, sut.localId)

        sut.localId = 1234567891011L
        assertEquals(1234567891011L, sut.localId)
    }
}
